doc/community/developer-guide.md

```diff
@@ -205,4 +205,96 @@ pytest src/tests/integration/test_generic_db_operations.py::test_profile_query
 
 We run integration tests against cloud databases like Snowflake, which requires using pre-registered accounts to evaluate their behavior. To initiate these tests, please create a branch in our [ploomber/jupyter repository](https://github.com/ploomber/jupysql).
 
-Please note that if you submit a pull request from a forked repository, the integration testing phase will be skipped because the pre-registered accounts won't be accessible.
\ No newline at end of file
+Please note that if you submit a pull request from a forked repository, the integration testing phase will be skipped because the pre-registered accounts won't be accessible.
+## General SQL Clause for Multiple Database Dialects
+
+### Context
+
+As our codebase is expanding, we have noticed that we need to write SQL queries for different database dialects such as MySQL, PostgreSQL, SQLite, and more. Writing and maintaining separate queries for each database can be time-consuming and error-prone.
+
+To address this issue, we can use `sqlglot` to create a construct that can be compiled across multiple SQL dialects. This clause will allow us to write a single SQL query that can be translated to different database dialects, then use it for calculating the metadata (e.g. metadata used by boxplot)
+
+In this document, we'll explain how to build generic SQL constructs and provide examples of how it can be used in our codebase. We will also include instructions on how to add support for additional database dialects.
+
+### Approach 1 - Provide the general SQL Clause
+
+We can use [SQLGlot](https://sqlglot.com/sqlglot.html) to build the general sql expressions.
+
+Then transpile to the sql which is supported by current connected dialect.
+
+Our `sql.connection.Connection._transpile_query` will automatically detect the dialect and transpile the SQL clause.
+
++++
+
+#### Example
+
+```{code-cell} ipython3
+from sqlglot import select, condition
+from sql import connection
+from sqlalchemy import create_engine
+
+# Prepare connection
+conn = connection.Connection(engine=create_engine(url="sqlite://"))
+
+where = condition("x=1").and_("y=1")
+general_sql = select("*").from_("y").where(where).sql()
+
+print("General SQL Clause: ")
+print(f"{general_sql}\n")
+
+print("Transpiled result: ")
+conn._transpile_query(general_sql)
```

Comment:
![image](https://user-images.githubusercontent.com/103026689/232440257-17958a80-12aa-41b4-8825-23a9c6c7e477.png)

I think this might be hard to read this output cell.

What do you think if we'll split it into 3 code cells, where the 1st cell initializes the connection, 2nd cell constructs the SQL clause, and 3rd cell transpiles the query. We can drop the prints and use headers above each cell.

Reply:
This is already implemented here?
https://github.com/ploomber/jupysql/pull/371#issuecomment-1511466470